purchase_product_weight/models/purchase.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2019 Sergio Corato <https://github.com/sergiocorato>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, fields, models
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = 'purchase.order.line'
+
+    weight_total = fields.Float(
+        string='Total weight',
+        help='Total weight computed from product weight * product quantity. '
+             'It can be written manually, in this case unit price will be '
+             'recomputed on Total weight / product quantity * seller price '
+             'if field "Compute price on weight" on product is flagged.')
+
+    @api.onchange('product_id', 'product_qty')
+    def _onchange_total_weight(self):
+        for line in self:
+            line.weight_total = line.product_id.weight * \
+                line.product_qty
```

Comment:
I think this function's name is `_onchange_weight_total` better than`_onchange_total_weight`. Do you think so?

Reply:
Right, done